lib/fakefs/file.rb

```diff
@@ -12,7 +12,24 @@ class File < StringIO
       APPEND_READ_WRITE   = 'a+'.freeze
     ].freeze
 
+    FMODE_READABLE = 0x00000001
+    FMODE_WRITABLE = 0x00000002
+    FMODE_READWRITE = (FMODE_READABLE | FMODE_WRITABLE)
+    FMODE_BINMODE = 0x00000004
+    FMODE_APPEND = 0x00000040
+    FMODE_CREATE = 0x00000080
+    FMODE_EXCL = 0x00000400
+    FMODE_TRUNC = 0x00000800
+    FMODE_TEXTMODE = 0x00001000
+
+    TEXT_MODES = {
+      'r' => FMODE_READABLE,
+      'w' => FMODE_WRITABLE | FMODE_TRUNC | FMODE_CREATE,
+      'a' => FMODE_WRITABLE | FMODE_APPEND | FMODE_CREATE
+    }.freeze
```

Comment:
where did you get these from ... some internal ruby place we can reference directly instead of copying ?

Reply:
I copied them from the ruby [header](https://github.com/ruby/ruby/blob/v2_7_8/include/ruby/io.h#L108). I doubt it's required to access them directly, these - unlike `O_SOMETHING` constants like `O_WRONLY`, aka `File::WRONLY` - are just bit flags well crafted for the purpose.
BTW, it looks like File constants are unnecessary and can be replaced with `FMODE`-s (but you cannot use `File::BINARY`, for example, it's just zero on linux, and ruby uses `FMODE_BINMODE` in `binmode?`)

Reply:
kk, added a inline comment